src/Init/Prelude.lean

```diff
@@ -995,94 +1088,184 @@ instance (r : α → γ → Sort u) : Trans Eq r r where
 instance (r : α → β → Sort u) : Trans r Eq r where
   trans h' heq := heq ▸ h'
 
+/-- The typeclass behind the notation `a + b : γ` where `a : α`, `b : β`. -/
 class HAdd (α : Type u) (β : Type v) (γ : outParam (Type w)) where
+  /-- The implementation of `a + b : γ`. -/
   hAdd : α → β → γ
 
+/-- The typeclass behind the notation `a - b : γ` where `a : α`, `b : β`. -/
 class HSub (α : Type u) (β : Type v) (γ : outParam (Type w)) where
+  /-- The implementation of `a - b : γ`. -/
   hSub : α → β → γ
 
+/-- The typeclass behind the notation `a * b : γ` where `a : α`, `b : β`. -/
 class HMul (α : Type u) (β : Type v) (γ : outParam (Type w)) where
+  /-- The implementation of `a * b : γ`. -/
   hMul : α → β → γ
 
+/-- The typeclass behind the notation `a / b : γ` where `a : α`, `b : β`. -/
 class HDiv (α : Type u) (β : Type v) (γ : outParam (Type w)) where
+  /-- The implementation of `a / b : γ`. -/
   hDiv : α → β → γ
 
+/-- The typeclass behind the notation `a % b : γ` where `a : α`, `b : β`. -/
 class HMod (α : Type u) (β : Type v) (γ : outParam (Type w)) where
+  /-- The implementation of `a % b : γ`. -/
   hMod : α → β → γ
 
+/-- The typeclass behind the notation `a ^ b : γ` where `a : α`, `b : β`. -/
 class HPow (α : Type u) (β : Type v) (γ : outParam (Type w)) where
+  /-- The implementation of `a ^ b : γ`. -/
   hPow : α → β → γ
 
+/-- The typeclass behind the notation `a ++ b : γ` where `a : α`, `b : β`. -/
 class HAppend (α : Type u) (β : Type v) (γ : outParam (Type w)) where
+  /-- The implementation of `a ++ b : γ`. -/
   hAppend : α → β → γ
 
+/--
+The typeclass behind the notation `a <|> b : γ` where `a : α`, `b : β`.
+Because `b` is "lazy" in this notation, it is passed as `Unit → β` to the
+implementation so it can decide when to evaluate it.
+-/
 class HOrElse (α : Type u) (β : Type v) (γ : outParam (Type w)) where
+  /-- The implementation of `a <|> b : γ`. `b` is passed as a thunk so it
+  can be forced only when needed. -/
   hOrElse : α → (Unit → β) → γ
 
+/--
+The typeclass behind the notation `a >> b : γ` where `a : α`, `b : β`.
+Because `b` is "lazy" in this notation, it is passed as `Unit → β` to the
+implementation so it can decide when to evaluate it.
+-/
 class HAndThen (α : Type u) (β : Type v) (γ : outParam (Type w)) where
+  /-- The implementation of `a >> b : γ`. `b` is passed as a thunk so it
+  can be forced only when needed. -/
   hAndThen : α → (Unit → β) → γ
 
+/-- The typeclass behind the notation `a &&& b : γ` where `a : α`, `b : β`. -/
 class HAnd (α : Type u) (β : Type v) (γ : outParam (Type w)) where
+  /-- The implementation of `a &&& b : γ`. -/
```

Comment:
Is this docstring shown when hovering over `a &&& b`?  If so, then I'd change the docstring to something more helpful including the intended meaning of the operation:
```suggestion
  /-- `a &&& b` computes the bitwise and of `a` and `b`. -/
```

Same goes for the other TC ops.

Reply:
It's an overloadable operation, so it might not actually do that. Certainly we should put the "intended meaning" of the operation in the docstring, maybe a couple important instances, but it's ultimately just an overloadable symbol.

Reply:
Yes, it is supposed to be bitwise and (and because it's an obscure hieroglyph it makes even more sense to explain that in the docstring).  All of the bitwise ops are triple characters to avoid clashes with `&&`, `||`, and `>>`.

> but it's ultimately just an overloadable symbol

Sure, nothing prevents you from providing weird instances, but the class is called `HAnd` (where the and stands for bitwise and).  And I don't think it's a good idea to overload the bitwise and symbol with other operations just because you can.  It's easy enough to define your own operators after all.